hail/src/main/scala/is/hail/expr/ir/functions/StringFunctions.scala

```diff
@@ -266,14 +257,16 @@ object StringFunctions extends RegistryFunctions {
             val n = cb.newLocal("hamming_n", 0)
             val i = cb.newLocal("hamming_i", 0)
 
-            val v1 = sc1.asBytes().memoize(cb, "hamming_bytes_1")
-            val v2 = sc2.asBytes().memoize(cb, "hamming_bytes_2")
+            val v1 = cb.newLocal[String]("hamming_str_1", sc1.loadString())
+            val v2 = cb.newLocal[String]("hamming_str_2", sc2.loadString())
 
-            val m = v1.loadLength().cne(v2.loadLength())
+            val l1 = cb.newLocal[Int]("hamming_len_1", v1.invoke[Int]("length"))
+            val l2 = cb.newLocal[Int]("hamming_len_2", v2.invoke[Int]("length"))
+            val m = l1.cne(l2)
 
             IEmitCode(cb, m, {
-              cb.whileLoop(i < v1.loadLength(), {
-                cb.ifx(v1.loadByte(i).cne(v2.loadByte(i)),
+              cb.whileLoop(i < l1, {
+                cb.ifx(v1.invoke[Int, Char]("charAt", i).toI.cne(v2.invoke[Int, Char]("charAt", i).toI),
```

Comment:
This is a behavioral change. The old version calculated the hamming distance on bytes this one does so on character sequences.

Reply:
I agree, the old implementation was wrong.